aws/src/integration/java/org/apache/iceberg/aws/glue/GlueCatalogLockTest.java

```diff
@@ -0,0 +1,88 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+package org.apache.iceberg.aws.glue;
+
+import java.util.List;
+import java.util.stream.Collectors;
+import java.util.stream.IntStream;
+import org.apache.iceberg.AppendFiles;
+import org.apache.iceberg.DataFile;
+import org.apache.iceberg.DataFiles;
+import org.apache.iceberg.PendingUpdate;
+import org.apache.iceberg.Table;
+import org.apache.iceberg.aws.AwsProperties;
+import org.apache.iceberg.aws.s3.S3FileIO;
+import org.apache.iceberg.catalog.TableIdentifier;
+import org.junit.AfterClass;
+import org.junit.Assert;
+import org.junit.BeforeClass;
+import org.junit.Test;
+import software.amazon.awssdk.services.dynamodb.DynamoDbClient;
+import software.amazon.awssdk.services.dynamodb.model.DeleteTableRequest;
+
+public class GlueCatalogLockTest extends GlueTestBase {
+
+  private static String lockTableName;
+  private static DynamoDbClient dynamo;
+
+  @BeforeClass
+  public static void beforeClass() {
+    GlueTestBase.beforeClass();
+    String testBucketPath = "s3://" + testBucketName + "/" + testPathPrefix;
+    lockTableName = getRandomName();
+    S3FileIO fileIO = new S3FileIO(clientFactory::s3);
+    glueCatalog = new GlueCatalog();
+    AwsProperties awsProperties = new AwsProperties();
+    awsProperties.setGlueCatalogLockTable(lockTableName);
+    awsProperties.setGlueCatalogLockEnabled(true);
+    dynamo = clientFactory.dynamo();
+    glueCatalog.initialize(catalogName, testBucketPath, awsProperties, glue,
+        new DynamoLockManager(dynamo, awsProperties), fileIO);
+  }
+
+  @AfterClass
+  public static void afterClass() {
+    GlueTestBase.afterClass();
+    dynamo.deleteTable(DeleteTableRequest.builder().tableName(lockTableName).build());
+  }
+
+  @Test
+  public void testParallelCommit() {
```

Comment:
You might want to just copy `testConcurrentFastAppends`. We know that does a good job of catching collisions. https://github.com/apache/iceberg/blob/master/hive-metastore/src/test/java/org/apache/iceberg/hive/TestHiveTableConcurrency.java#L45-L79